homeassistant/components/sensor/trafikverket_weatherstation.py

```diff
@@ -0,0 +1,129 @@
+"""
+Weather information for air and road temperature, provided by Trafikverket.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.trafikverket_weatherstation/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME, ATTR_ATTRIBUTION, TEMP_CELSIUS)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ATTRIBUTION = "Data provided by Trafikverket API"
+
+CONF_API = 'api'
+CONF_STATION = 'station'
+CONF_TYPE = 'type'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+SCAN_INTERVAL = timedelta(seconds=300)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_API): cv.string,
+    vol.Required(CONF_STATION): cv.string,
```

Comment:
Do you have a list of valid stations?

Reply:
Yes, it's written in the documentation under header "See Trafikverket weather stations":
https://github.com/home-assistant/home-assistant.github.io/pull/4562/files

In the map you will need to click a choosen station and then copy it's name, like highlighted:
https://imgur.com/a/TGQKC

Thanks for your review and notice of the other issues. I will try to fix them tomorrow or during the weekend. Br Qxlkdr

Reply:
So what happens if the user provides an invalid station or api token?

Reply:
I will try to fix that. However, do I need to have an error handler for both cases? Or can I just check for a int/double? If the result variable doesn't contains an int/double the has been an error in the request. As I see it, it could only be 3 cases:
1. No network
2. Invalid station
3. Invalid API key

I can have an error handler for number 2-3 (checking for int/double). If not received an error message like "Cannot fetch station, check station name and API key" in the log. Number 1 can be checked separately. So, 3 issues, 2 solutions to all 3 of them. Is that ok or will I need to have an error handler for all 3, separately?

Reply:
The most important is that the code does not crash in any of these cases.
And it would be great if you can provide a log message to help the user fix the problem.